onnx2torch/node_converters/pad.py

```diff
@@ -0,0 +1,126 @@
+__all__ = [
+    'OnnxPadStatic',
+    'OnnxPadDynamic',
+]
+
+from typing import List
+from typing import Optional
+
+import torch
+import torch._C as torch_C
+from torch import nn
+
+from onnx2torch.node_converters.registry import add_converter
+from onnx2torch.onnx_graph import OnnxGraph
+from onnx2torch.onnx_node import OnnxNode
+from onnx2torch.utils.common import OnnxMapping
+from onnx2torch.utils.common import OnnxToTorchModule
+from onnx2torch.utils.common import OperationConverterResult
+from onnx2torch.utils.common import onnx_mapping_from_node
+from onnx2torch.utils.custom_export_to_onnx import CustomExportToOnnx
+from onnx2torch.utils.custom_export_to_onnx import OnnxToTorchModuleWithCustomExport
+
+
+class OnnxPadStatic(nn.Module, OnnxToTorchModule):
+
+    def __init__(
+        self,
+        pads: List[int],
+        mode: str = 'constant',
+        constant_value: float = 0.0,
+    ):
+        super().__init__()
+        self.mode = mode
+        self.pads = pads
+        self.constant_value = constant_value
+
+    def forward(self, input_tensor: torch.Tensor) -> torch.Tensor:
+        return torch.nn.functional.pad(
+            input_tensor,
+            mode=self.mode,
+            pad=self.pads,
+            value=self.constant_value,
+        )
+
+
+class OnnxPadDynamic(nn.Module, OnnxToTorchModuleWithCustomExport):
+
+    def __init__(self, mode: str = 'constant'):
+        super().__init__()
+        self.mode = mode
+
+    @staticmethod
+    def _do_forward(
+        input_tensor: torch.Tensor,
+        mode: str,
+        pads: torch.Tensor,
+        constant_value: Optional[float],
+    ) -> torch.Tensor:
+
+        if constant_value is not None:
+            return torch.nn.functional.pad(input_tensor, mode=mode, pad=pads, value=constant_value)
+
+        return torch.nn.functional.pad(input_tensor, mode=mode, pad=pads)
+
+    def forward(
+        self,
+        input_tensor: torch.Tensor,
+        pads: torch.Tensor,
+        constant_value: Optional[float] = None,
+    ) -> torch.Tensor:
+
+        if torch.unique(pads).shape[0] > 1:
+            raise NotImplementedError(f'Only uniform padding on all axes is implemented ({pads})')
+
+        pads = tuple(pads.tolist())
+        output = self._do_forward(input_tensor, self.mode, pads, constant_value)
```

Comment:
In `_do_forward` `pads` typing is `torch.Tensor`, but why do you cast it to tuple? 
In below you cast it back to torch.Tensor.
Do less casting.
Like (it may be wrong, check):
```python
# i delete tuple casting in line #75, and make "inplace" casting and you dont need cas it back to tensor while onnx export
# dont forget change casting for `pad`
output = self._do_forward(input_tensor, self.mode, tuple(pads), constant_value)
if torch.onnx.is_in_onnx_export():
            args = [self.mode, input_tensor, pads]
```

Reply:
Done